src/swish/http.ss

```diff
@@ -462,7 +467,7 @@
               (raise 'content-limit-exceeded))
             (let ([content (make-bytevector len)])
               (get-chunk! ip content len)
-              (values (parse-encoded-kv content 0 len) #f))]
+              (values (parse-encoded-kv content 0 len) pos))]
```

Comment:
Since we just read exactly len bytes, I don't see a need to return pos here.

Reply:
Returning `pos` lets us detect another case where the handler reads too much. See line 277 of http.ms. If the handler behaves, `advance!` is pretty low overhead.

Aside: perhaps we should pull the `eof-object?` test outside the loop in `advance!`:
```
(do ([n remaining (fx- n 1)] [b #f (get-u8 ip)])
    ((fx= n 0)
     (when (eof-object? b)
       (raise 'unexpected-eof))))
```

Reply:
OK, I'll put the pos back in there. I chose not to pull the eof-object? test because it's very quick and this code should not run at all in the common case.